crates/guest-rust/src/lib.rs

```diff
@@ -250,7 +250,18 @@ impl<T: WasmResource> Resource<T> {
     where
         T: RustResource,
     {
-        let _ = Box::from_raw(rep as *mut T);
+        let _ = Box::from_raw(rep as *mut Option<T>);
+    }
+
+    /// Takes back ownership of the object, dropping the resource handle.
+    pub fn take(resource: Self) -> T
```

Comment:
Could this be rename to `into_inner` perhaps? The `take` terminology seems derivitive of `Option<T>` which is ideally only an implementation detail here.